ecl/hqlcpp/hqlcppds.cpp

```diff
@@ -3236,6 +3236,60 @@ class InlineDatasetSkipCallback : public CInterface, implements IHqlCodeCallback
     }
 };
 
+void HqlCppTranslator::buildDatasetAssignCombine(BuildCtx & ctx, IHqlCppDatasetBuilder * target, IHqlExpression * expr)
+{
+    BuildCtx iterctx(ctx);
+    IHqlExpression *left = expr->queryChild(0);
+    IHqlExpression *right = expr->queryChild(1);
+
+    iterctx.addGroup();  // stop bound cursors leaking outside the block.
+
+    // right iter1; right.first();
+    HqlExprAttr rightIter, rightRow;
+    Owned<IHqlCppDatasetCursor> cursor = createDatasetSelector(iterctx, right);
+    cursor->buildIterateClass(iterctx, rightIter, rightRow);
+    buildIteratorFirst(iterctx, rightIter, rightRow);
+    BoundRow * sourceCursor = buildDatasetIterate(iterctx, left, false);
+    if (!sourceCursor)
+        return;
+
+    // handle right rows < left rows failure
+    BuildCtx failctx(iterctx);
+    StringBuffer s("if (!");
```

Comment:
I am generally trying to move away from any explicit c++ code.  So I would code this as

OwnedHqlExpr test = createValue(no_not, makeBoolType(), LINK(rightRow));
iterctx.addFilter(test);

This implicitly selects within the nested block 

similarly for the fail I would create a no_fail expression, and call buildStmt(), rather than directly calling the function.